### ISSUE TITLE
Alarm Server & GUI don't update list of PVs live

### ISSUE THREAD

**Issue by IsabellaRey** [ISISComputingGroup/IBEX#1723]
Steps to reproduce:
1. Start instrument and GUI
2. Setup a config with the Eurotherm IOC running only with ADDR 01
3. Start the Eurotherm IOC
4. The Alarms perspective lists a few PVs from EUROTHERM1:A01
5. Stop the Eurotherm IOC
6. Change the config macros to have also ADDR 02
7. Restart the Eurotherm IOC
8. The alarm perspective still only displays PVs from EUROTHERM1:A01, not A02

(this happens also the other way round, e.g. I start with two addresses and then switch to one --> I still see the PVs from A01 and A02 instead of just one of them)

Restarting the GUI alone or the Alarm Server alone doesn't help, they have to be both restarted

Acceptance:

1. Config update is automatically applied.

**Comment by IsabellaRey** [comment]
NOTE on my machine I keep getting alarms about the Eurotherm PVs even after (a loooot of) fiddling with the config directory (completely cleared and cloned from scratched), a lot of instrument restarting, and pointing at a config which is completely empty

**Comment by John-Holt-Tessella** [comment]
When done update manual system test 4-2 to highlight what an appropriate error tree would look like

**Comment by kjwoodsISIS** [comment]
Check if the alarm can be refreshed with a simple command (i.e. re-run startup mechanism)/
If that works - good.  If not, create a new ticket to investigate further.

**Comment by Tom-Willemsen** [comment]
Blocked by #2150

**Comment by Tom-Willemsen** [comment]
There are two seperate issues here:

- The alarm server doesn't keep the MySQL database up to date properly. If you restart the alarm server the correct values *do* get inserted into the DB. I am currently looking for a way to force the alarm server to restart (or otherwise perform the process which inserts correct values into the DB) whenever a configuration is changed.

- The GUI doesn't keep up to date with the changes in the MySQL database. This can be fixed by restarting the GUI or by switching to "annunciator" and then back to "instrument" within the alarms perspective. It should be possible from within java to tell the perspective to reload all data from the model, so I will investigate doing that.

**Comment by Tom-Willemsen** [comment]
Two pull requests for this issue:
https://github.com/ISISComputingGroup/ibex_gui/pull/476
https://github.com/ISISComputingGroup/EPICS-inst_servers/pull/117

**Comment by John-Holt-Tessella** [comment]
Check that alarms are for running IOCs not just for all IOCs. If this fails we may need to pass a flag to the make alarm server config file.

**Comment by John-Holt-Tessella** [comment]
When merged advise people to update there configs using the upgrade script.

**Comment by Tom-Willemsen** [comment]
Sorry, the review comments are scattered accross a few of the PRs. Generally works ok, just a few things to address before this can be merged.

**Comment by AdrianPotter** [comment]
- [ ] Please let Dominic and Alexandra on HRPD know when this is done. They specifically flagged the alarm view as something they'd like to be better
